File: Data Structures & Algorithms/longest-strictly-increasing-or-strictly-decreasing-subarray/submission-2.java
class Solution {
    public int longestMonotonicSubarray(int[] nums) {
        boolean in = false,de = false;
        int cnt = 1,ans = 1;
        for(int i = 1; i < nums.length; i++)
        {
            if(!in && !de)
            {
                if(nums[i] > nums[i - 1])
                {
                    in = true;
                    cnt+=1;
                }
                else if(nums[i] < nums[i - 1])
                {
                    de = true;
                    cnt+=1;
                }
            }
            else if(in && nums[i] > nums[i - 1]) cnt+=1;
            else if(de && nums[i] < nums[i - 1]) cnt+=1;
            else if(in && nums[i] < nums[i - 1])
            {
                in = false;
                de = true;
                cnt = 2;
            }
            else if(de && nums[i] > nums[i - 1])
            {
                de = false;
                in = true;
                cnt = 2;
            }
            else
            {
                in = false;
                de = false;
                cnt = 1;
            }
            ans = Math.max(ans,cnt);
        }
        return ans;
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if(nums.size() == 0) return 0;
        map<int,int>mp;
        for(int i : nums) mp[i]++;
        int lst = INT_MAX;
        int ans = 1;
        int cnt = 1;
        for(auto it : mp)
        {
            if(lst == INT_MAX)
            {
                lst = it.first;
                continue;
            }
            if(it.first - lst == 1) cnt+=1;
            else cnt = 1;
            lst = it.first;
            ans = max(cnt,ans);
        }
        return ans;
    }
};
